src/callbacks_step/amr_dg2d.jl

```diff
@@ -1,3 +1,95 @@
+# Redistribute data for load balancing after partitioning the mesh
+function redistribute_data!(u_ode::AbstractVector, mesh::TreeMesh{2}, equations, 
+                            dg::DGSEM, cache, old_mpi_ranks_per_cell)
+  # Retain current solution data
+  old_n_elements = nelements(dg, cache)
+  old_cell_ids = copy(cache.elements.cell_ids)
+  old_u_ode = copy(u_ode)
+  GC.@preserve old_u_ode begin # OBS! If we don't GC.@preserve old_u_ode, it might be GC'ed
+    old_u = wrap_array(old_u_ode, mesh, equations, dg, cache)
+
+    reinitialize_containers!(mesh, equations, dg, cache)
+    @unpack elements = cache
+
+    resize!(u_ode, nvariables(equations) * nnodes(dg)^ndims(mesh) * nelements(dg, cache))
+    u = wrap_array(u_ode, mesh, equations, dg, cache)
+
+    # Get new list of leaf cells
+    leaf_cell_ids = local_leaf_cells(mesh.tree)
+
+    # Collect MPI requests for MPI_Waitall
+    reqs = Vector{MPI.Request}()
+
+    # Find elements that will change their rank and send their data to the new rank
+    for old_element_id in 1:old_n_elements
+      cell_id = old_cell_ids[old_element_id]
+      if !(cell_id in leaf_cell_ids)
+        # Send element data to new rank, use cell_id as tag (non-blocking)
+        dest = mesh.tree.mpi_ranks[cell_id]
+        req = MPI.Isend(vec(old_u[:, .., old_element_id]), dest, cell_id, mpi_comm())
```

Comment:
Just to be sure I understand it correctly: In the `Isend` we need `vec`, but in the `Irecv` below we can just pass a view to the multi-dimensional target array?

Reply:
We don't need it in `Isend`, you're right.